kotlin/internal/jvm/impl.bzl

```diff
@@ -156,14 +155,33 @@ def kt_jvm_binary_impl(ctx):
         ),
     )
 
+_SPLIT_STRINGS = [
+    "src/test/java/",
+    "javatests/",
+    "java/",
+    "test/",
+]
+
 def kt_jvm_junit_test_impl(ctx):
     providers = _kt_jvm_produce_jar_actions(ctx, "kt_jvm_test")
     runtime_jars = depset(ctx.files._bazel_test_runner, transitive = [providers.java.transitive_runtime_jars])
+
+    test_class = ctx.attr.test_class
+
+    # If no test_class, do a best-effort attempt to infer one.
+    if not bool(ctx.attr.test_class):
+        for file in ctx.files.srcs:
+            if file.basename.split(".")[0] == ctx.attr.name:
+                for splitter in _SPLIT_STRINGS:
+                    elements = file.short_path.split(splitter, maxsplit = 1)
```

Comment:
Add some examples here on the string manipulation inputs and expected outputs?

Reply:
Not sure what you mean. Like a comment including examples like `common/foo/src/test/java/com/squareup/foo/TestFoo.kt` turning into `["common/foo", "com/squareup/foo/TestFoo.kt"]` ?

Reply:
Nvm, don't worry about it.